### PR TITLE
chore(feat): Add kubelet service kill experiment in generic experiment list

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: Udit Gaurav <udit.gaurav@mayadata.io>

**What this PR does / why we need it**:
- This PR is for enhancing the infra chaos of litmus with the addition of service chaos experiment in the list of generic experiments. A `kubelet` service kill experiment has been added in this PR.
- The main logic behind the service kill experiment will be stopping the service gracefully for a certain duration and then restarting it after the total chaos duration. This is done with the help of Daemonsets that have the root access of the nodes.

- Tested Platforms:
  - [X] GKE
  - [X] EKS
  - [X] OpenShift

**Which issue this PR fixes**
- fixes https://github.com/litmuschaos/litmus/issues/1541

**Checklist**

* [X] Does this PR have a corresponding GitHub issue?
* [X] Have you included relevant README for the chaoslib/experiment with details?
* [X] Have you added debug messages where necessary? 
* [X] Have you added task comments where necessary? 
* [X] Have you tested the changes for possible failure conditions?
* [X] Have you provided the positive & negative test logs for the litmusbook execution?
* [X] Does the litmusbook ensure idempotency of cluster state?, i.e., is cluster restored to original state?
* [ ] Have you used non-shell/command modules for Kubernetes tasks?
* [X] Have you (jinja) templatized custom scripts that is run by the litmusbook, if any? 
* [X] Have you (jinja) templatized Kubernetes deployment manifests used by the litmusbook, if any?
* [X] Have you reused/created util functions instead of repeating tasks in the litmusbook?
* [X] Do the artifacts follow the appropriate directory structure? 
* [ ] Have you isolated storage (eg: OpenEBS) specific implementations, checks? 
* [X] Have you isolated platform (eg: baremetal kubeadm/openshift/aws/gcloud) specific implementations, checks?  
* [X] Are the ansible facts well defined? Is the scope explicitly set for playbook & included utils?
* [ ] Have you ensured minimum/careful usage of shell utilities (awk, grep, sed, cut, xargs etc.,)?
* [ ] Can the limtusbook be executed both from within & outside a container (configurable paths, no hardcode)?
* [X] Can you suggest the minimal resource requirements for the litmusbook execution?
* [X] Does the litmusbook job artifact carry comments/default options/range for the ENV tunables?
* [X] Has the litmusbooks been linted? 

**Special notes for your reviewer**:
- In place of helper pod, we use in almost all experiments in this experiment, the role of helper pod is played by Daemonset. Daemonset provides the host access to the nodes and contains the main logic of the chaos experiment. We are doing an exec to the daemonset pod and performing some operations.

### CODE REVIEW

**Comment by uditgaurav** [comment]
**Node doing down due to kubelet kill:**
```bash
NAME                                         STATUS     ROLES    AGE   VERSION
ip-172-31-30-10.us-east-2.compute.internal   NotReady   <none>   19d   v1.15.11-eks-af3caf
ip-172-31-32-48.us-east-2.compute.internal   Ready      <none>   19d   v1.15.11-eks-af3caf
```
**Job pod and Daemonsets**
```bash
NAME                                 READY   STATUS    RESTARTS   AGE
chaos-operator-ce-5cd5894879-r2n29   1/1     Running   0          67m
kubelet-service-kill-dttdq-r5mvk     1/1     Running   0          102s
nginx-7bb7cd8db5-szvwq               1/1     Running   0          67m
service-kill-ds-mffvbq-95nqw         1/1     Running   0          94s
service-kill-ds-mffvbq-c6t56         1/1     Running   0          94s
```
_Logs of the experiment:_

```
ansible-playbook 2.7.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 2.7.15+ (default, Oct  7 2019, 17:39:04) [GCC 7.4.0]
Using /etc/ansible/ansible.cfg as config file
/etc/ansible/hosts did not meet host_list requirements, check plugin documentation if this is unexpected
/etc/ansible/hosts did not meet script requirements, check plugin documentation if this is unexpected
statically imported: /experiments/generic/kubelet_service_kill/kubelet_service_kill_ansible_prerequisites.yml

2020-06-09 16:44:04.681326 PLAYBOOK: kubelet_service_kill_ansible_logic.yml 
1 plays in ./experiments/generic/kubelet_service_kill/kubelet_service_kill_ansible_logic.yml

2020-06-09 16:44:04.695359 ************ BRACE YOURSELF, EXPERIMENT BEGINS! ************ 

2020-06-09 16:44:05.384673 Step: Gathering Facts 
task path: /experiments/generic/kubelet_service_kill/kubelet_service_kill_ansible_logic.yml:2
META: ran handlers

2020-06-09 16:44:05.850155 Step: [PreReq] Identify the chaos util for kubelet-service-kill experiment 
task path: /experiments/generic/kubelet_service_kill/kubelet_service_kill_ansible_prerequisites.yml:1
details => {"changed": true, "checksum": "2d76d919fa3c20d76eff0b8e3148897f4319c491", "dest": "/tmp/chaosutil.yml", "gid": 0, "group": "root", "md5sum": "c855258e0f920687b5ce930bafa965a7", "mode": "0644", "owner": "root", "size": 73, "src": "/tmp/ansible-tmp-1591721045.44-98732516724457/source", "state": "file", "uid": 0}

2020-06-09 16:44:05.915248 Step: [PreReq]: Including the chaos util for the kubelet-service-kill experiment 
task path: /experiments/generic/kubelet_service_kill/kubelet_service_kill_ansible_logic.yml:28
details => {"ansible_facts": {"c_util": "/chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml"}, "ansible_included_var_files": ["/tmp/chaosutil.yml"], "changed": false}
included: /utils/runtime/update_chaos_result_resource.yml for 127.0.0.1

2020-06-09 16:44:06.233452 Step: [PreReq]: Generate the chaos result CR to reflect SOT (Start of Test) 
task path: /utils/runtime/update_chaos_result_resource.yml:3
details => {"changed": true, "checksum": "c8d58ff872d1de2e341f3c976653b9fca6563b3b", "dest": "/tmp/chaos-result.yml", "gid": 0, "group": "root", "md5sum": "ef1be1ec3abaf3e192d4ca2d0855b73b", "mode": "0644", "owner": "root", "size": 276, "src": "/tmp/ansible-tmp-1591721046.06-179170004444122/source", "state": "file", "uid": 0}

2020-06-09 16:44:06.827884 Step: [PreReq]: Apply the chaos result CR for kubelet-service-kill experiment 
task path: /utils/runtime/update_chaos_result_resource.yml:13
details => {"changed": true, "cmd": "kubectl apply -f /tmp/chaos-result.yml -n litmus", "delta": "0:00:00.379052", "end": "2020-06-09 16:44:06.808170", "failed_when_result": false, "rc": 0, "start": "2020-06-09 16:44:06.429118", "stderr": "", "stderr_lines": [], "stdout": "chaosresult.litmuschaos.io/kubelet-service-kill configured", "stdout_lines": ["chaosresult.litmuschaos.io/kubelet-service-kill configured"]}

2020-06-09 16:44:06.921133 Step: [Info]: Display the application information passed via the test job 
task path: /experiments/generic/kubelet_service_kill/kubelet_service_kill_ansible_logic.yml:46
details => {
    "msg": [
        "The application info is as follows:", 
        "Namespace    : litmus", 
        "Label        : run=nginx", 
        "Ramp Time    : "
    ]
}
included: /utils/common/status_app_pod.yml for 127.0.0.1

2020-06-09 16:44:07.202828 Step: [Status]: Checking whether application pods are in running state 
task path: /utils/common/status_app_pod.yml:2
details => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -n litmus -l run=nginx -o custom-columns=:.status.phase --no-headers", "delta": "0:00:00.110973", "end": "2020-06-09 16:44:07.182113", "rc": 0, "start": "2020-06-09 16:44:07.071140", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}

2020-06-09 16:44:07.453968 Step: [Status]: Checking whether application containers are in running state 
task path: /utils/common/status_app_pod.yml:9
details => {"attempts": 1, "changed": true, "cmd": "kubectl get pod -n litmus -l run=nginx --no-headers -o jsonpath='{.items[*].status.containerStatuses[*].ready}' | tr ' ' '\\n' | uniq", "delta": "0:00:00.113275", "end": "2020-06-09 16:44:07.434108", "rc": 0, "start": "2020-06-09 16:44:07.320833", "stderr": "", "stderr_lines": [], "stdout": "true", "stdout_lines": ["true"]}
included: /chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml for 127.0.0.1

2020-06-09 16:44:07.840264 Step: [Prepare]: Select the application pod name 
task path: /chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml:6
details => {"changed": true, "cmd": "kubectl get pod -l run=nginx -n litmus -o=custom-columns=:metadata.name --no-headers | shuf | head -1", "delta": "0:00:00.114487", "end": "2020-06-09 16:44:07.823537", "rc": 0, "start": "2020-06-09 16:44:07.709050", "stderr": "", "stderr_lines": [], "stdout": "nginx-7bb7cd8db5-szvwq", "stdout_lines": ["nginx-7bb7cd8db5-szvwq"]}

2020-06-09 16:44:07.881663 Step: [Prepare]: Recording the application pod name 
task path: /chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml:15
details => {"ansible_facts": {"app_pod": "nginx-7bb7cd8db5-szvwq"}, "changed": false}

2020-06-09 16:44:08.121286 Step: [Prepare]: Identify the application node name 
task path: /chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml:21
details => {"changed": true, "cmd": "kubectl get pod nginx-7bb7cd8db5-szvwq -n litmus --no-headers -o custom-columns=:spec.nodeName", "delta": "0:00:00.112883", "end": "2020-06-09 16:44:08.104302", "rc": 0, "start": "2020-06-09 16:44:07.991419", "stderr": "", "stderr_lines": [], "stdout": "ip-172-31-30-10.us-east-2.compute.internal", "stdout_lines": ["ip-172-31-30-10.us-east-2.compute.internal"]}

2020-06-09 16:44:08.158851 Step: [Prepare]: Record the application node name 
task path: /chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml:29
details => {"ansible_facts": {"app_node": "ip-172-31-30-10.us-east-2.compute.internal"}, "changed": false}

2020-06-09 16:44:08.320827 Step: [Prepare:] Generate a run id if not passed from the engine/experiment 
task path: /chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml:34
details => {"changed": true, "cmd": "echo $(mktemp) | cut -d '.' -f 2 | cut -c -6", "delta": "0:00:00.034673", "end": "2020-06-09 16:44:08.304523", "rc": 0, "start": "2020-06-09 16:44:08.269850", "stderr": "", "stderr_lines": [], "stdout": "mFfVBQ", "stdout_lines": ["mFfVBQ"]}

2020-06-09 16:44:08.360918 Step: set_fact 
task path: /chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml:38
details => {"ansible_facts": {"run_id": "mffvbq"}, "changed": false}

2020-06-09 16:44:08.568790 Step: [Prepare]: Patch the run_id to kubelet service kill daemonset template 
task path: /chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml:42
details => {"changed": true, "checksum": "1fac9713a90b5b029ac072a1cd413d8fc941dd2f", "dest": "/tmp/kubelet-service-kill-ds.yml", "gid": 0, "group": "root", "md5sum": "a4db659c2ec3cb7643f6a879390e8fe4", "mode": "0644", "owner": "root", "size": 875, "src": "/tmp/ansible-tmp-1591721048.4-45912449653880/source", "state": "file", "uid": 0}

2020-06-09 16:44:08.956234 Step: [Prepare]: Create kubelet service kill Daemonset 
task path: /chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml:47
details => {"changed": true, "cmd": "kubectl create -f /tmp/kubelet-service-kill-ds.yml -n litmus", "delta": "0:00:00.237113", "end": "2020-06-09 16:44:08.938954", "rc": 0, "start": "2020-06-09 16:44:08.701841", "stderr": "", "stderr_lines": [], "stdout": "daemonset.extensions/service-kill-ds-mffvbq created", "stdout_lines": ["daemonset.extensions/service-kill-ds-mffvbq created"]}
FAILED - RETRYING: [Prepare]: Confirm that the ds pods are running on all nodes (90 retries left).

2020-06-09 16:44:11.431812 Step: [Prepare]: Confirm that the ds pods are running on all nodes 
task path: /chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml:54
details => {"attempts": 2, "changed": true, "cmd": "kubectl get pod -l name=service-kill-ds-mffvbq --no-headers -o custom-columns=:status.phase -n litmus | sort | uniq", "delta": "0:00:00.121481", "end": "2020-06-09 16:44:11.408610", "rc": 0, "start": "2020-06-09 16:44:11.287129", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}

2020-06-09 16:44:11.688308 Step: [Prepare]: Derive the ds pod name on the application node 
task path: /chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml:65
details => {"changed": true, "cmd": "kubectl get pods -n litmus -l name=service-kill-ds-mffvbq --no-headers -o wide --field-selector spec.nodeName=ip-172-31-30-10.us-east-2.compute.internal | awk '{print$1}'", "delta": "0:00:00.119903", "end": "2020-06-09 16:44:11.669424", "rc": 0, "start": "2020-06-09 16:44:11.549521", "stderr": "", "stderr_lines": [], "stdout": "service-kill-ds-mffvbq-95nqw", "stdout_lines": ["service-kill-ds-mffvbq-95nqw"]}

2020-06-09 16:44:11.728626 Step: [Prepare]: Record the ds pod name 
task path: /chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml:73
details => {"ansible_facts": {"ds_pod": "service-kill-ds-mffvbq-95nqw"}, "changed": false}
included: /chaoslib/litmus/kubelet_service_kill/service_chaos.yml for 127.0.0.1

2020-06-09 16:44:12.228077 Step: [Inject]: Killing kubelet service for 90 seconds 
task path: /chaoslib/litmus/kubelet_service_kill/service_chaos.yml:13
details => {"changed": true, "cmd": "timeout 10 kubectl exec service-kill-ds-mffvbq-95nqw -n litmus -- bash -c \"systemctl stop kubelet && sleep 90 && systemctl start kubelet\"", "delta": "0:00:00.273953", "end": "2020-06-09 16:44:12.210826", "rc": 0, "start": "2020-06-09 16:44:11.936873", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
FAILED - RETRYING: [Status]: Waiting for node to get in NotReady state (90 retries left).
FAILED - RETRYING: [Status]: Waiting for node to get in NotReady state (89 retries left).
FAILED - RETRYING: [Status]: Waiting for node to get in NotReady state (88 retries left).
FAILED - RETRYING: [Status]: Waiting for node to get in NotReady state (87 retries left).
FAILED - RETRYING: [Status]: Waiting for node to get in NotReady state (86 retries left).
FAILED - RETRYING: [Status]: Waiting for node to get in NotReady state (85 retries left).
FAILED - RETRYING: [Status]: Waiting for node to get in NotReady state (84 retries left).
FAILED - RETRYING: [Status]: Waiting for node to get in NotReady state (83 retries left).
FAILED - RETRYING: [Status]: Waiting for node to get in NotReady state (82 retries left).
FAILED - RETRYING: [Status]: Waiting for node to get in NotReady state (81 retries left).
FAILED - RETRYING: [Status]: Waiting for node to get in NotReady state (80 retries left).
FAILED - RETRYING: [Status]: Waiting for node to get in NotReady state (79 retries left).
FAILED - RETRYING: [Status]: Waiting for node to get in NotReady state (78 retries left).
FAILED - RETRYING: [Status]: Waiting for node to get in NotReady state (77 retries left).
FAILED - RETRYING: [Status]: Waiting for node to get in NotReady state (76 retries left).

2020-06-09 16:44:46.377119 Step: [Status]: Waiting for node to get in NotReady state 
task path: /chaoslib/litmus/kubelet_service_kill/service_chaos.yml:18
details => {"attempts": 16, "changed": true, "cmd": "kubectl get nodes ip-172-31-30-10.us-east-2.compute.internal --no-headers -o custom-columns=STATUS:status.conditions[3].status", "delta": "0:00:00.118992", "end": "2020-06-09 16:44:46.355934", "rc": 0, "start": "2020-06-09 16:44:46.236942", "stderr": "", "stderr_lines": [], "stdout": "Unknown", "stdout_lines": ["Unknown"]}

2020-06-09 16:46:16.736549 Step: [Wait]: Wait for the chaos duration of 90s 
task path: /chaoslib/litmus/kubelet_service_kill/service_chaos.yml:28
details => {"changed": false, "elapsed": 90, "path": null, "port": null, "search_regex": null, "state": "started"}

2020-06-09 16:46:16.995501 Step: [Status]: Checking the node status after chaos 
task path: /chaoslib/litmus/kubelet_service_kill/service_chaos.yml:32
details => {"attempts": 1, "changed": true, "cmd": "kubectl get nodes ip-172-31-30-10.us-east-2.compute.internal --no-headers -o custom-columns=STATUS:status.conditions[3].status", "delta": "0:00:00.118176", "end": "2020-06-09 16:46:16.973506", "rc": 0, "start": "2020-06-09 16:46:16.855330", "stderr": "", "stderr_lines": [], "stdout": "True", "stdout_lines": ["True"]}

2020-06-09 16:46:17.259678 Step: [CleanUP]: Tear down service kill infra 
task path: /chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml:80
details => {"changed": true, "cmd": "kubectl delete ds -n litmus -l name=service-kill-ds-mffvbq", "delta": "0:00:00.133591", "end": "2020-06-09 16:46:17.241863", "rc": 0, "start": "2020-06-09 16:46:17.108272", "stderr": "", "stderr_lines": [], "stdout": "daemonset.extensions \"service-kill-ds-mffvbq\" deleted", "stdout_lines": ["daemonset.extensions \"service-kill-ds-mffvbq\" deleted"]}
FAILED - RETRYING: [Status]: Confirm that the svc chaos ds is stopped on all nodes (90 retries left).
FAILED - RETRYING: [Status]: Confirm that the svc chaos ds is stopped on all nodes (89 retries left).
FAILED - RETRYING: [Status]: Confirm that the svc chaos ds is stopped on all nodes (88 retries left).
FAILED - RETRYING: [Status]: Confirm that the svc chaos ds is stopped on all nodes (87 retries left).
FAILED - RETRYING: [Status]: Confirm that the svc chaos ds is stopped on all nodes (86 retries left).
FAILED - RETRYING: [Status]: Confirm that the svc chaos ds is stopped on all nodes (85 retries left).
FAILED - RETRYING: [Status]: Confirm that the svc chaos ds is stopped on all nodes (84 retries left).
FAILED - RETRYING: [Status]: Confirm that the svc chaos ds is stopped on all nodes (83 retries left).
FAILED - RETRYING: [Status]: Confirm that the svc chaos ds is stopped on all nodes (82 retries left).
FAILED - RETRYING: [Status]: Confirm that the svc chaos ds is stopped on all nodes (81 retries left).
FAILED - RETRYING: [Status]: Confirm that the svc chaos ds is stopped on all nodes (80 retries left).
FAILED - RETRYING: [Status]: Confirm that the svc chaos ds is stopped on all nodes (79 retries left).
FAILED - RETRYING: [Status]: Confirm that the svc chaos ds is stopped on all nodes (78 retries left).
FAILED - RETRYING: [Status]: Confirm that the svc chaos ds is stopped on all nodes (77 retries left).
FAILED - RETRYING: [Status]: Confirm that the svc chaos ds is stopped on all nodes (76 retries left).
FAILED - RETRYING: [Status]: Confirm that the svc chaos ds is stopped on all nodes (75 retries left).
FAILED - RETRYING: [Status]: Confirm that the svc chaos ds is stopped on all nodes (74 retries left).
FAILED - RETRYING: [Status]: Confirm that the svc chaos ds is stopped on all nodes (73 retries left).
FAILED - RETRYING: [Status]: Confirm that the svc chaos ds is stopped on all nodes (72 retries left).

2020-06-09 16:47:00.713440 Step: [Status]: Confirm that the svc chaos ds is stopped on all nodes 
task path: /chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml:87
details => {"attempts": 20, "changed": true, "cmd": "kubectl get pod -l name=service-kill-ds-mffvbq --no-headers -o custom-columns=:status.phase -n litmus | sort | uniq", "delta": "0:00:00.143866", "end": "2020-06-09 16:47:00.691628", "rc": 0, "start": "2020-06-09 16:47:00.547762", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
included: /utils/common/status_app_pod.yml for 127.0.0.1

2020-06-09 16:47:01.067895 Step: [Status]: Checking whether application pods are in running state 
task path: /utils/common/status_app_pod.yml:2
details => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -n litmus -l run=nginx -o custom-columns=:.status.phase --no-headers", "delta": "0:00:00.130582", "end": "2020-06-09 16:47:01.046664", "rc": 0, "start": "2020-06-09 16:47:00.916082", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}

2020-06-09 16:47:01.316691 Step: [Status]: Checking whether application containers are in running state 
task path: /utils/common/status_app_pod.yml:9
details => {"attempts": 1, "changed": true, "cmd": "kubectl get pod -n litmus -l run=nginx --no-headers -o jsonpath='{.items[*].status.containerStatuses[*].ready}' | tr ' ' '\\n' | uniq", "delta": "0:00:00.114453", "end": "2020-06-09 16:47:01.296075", "rc": 0, "start": "2020-06-09 16:47:01.181622", "stderr": "", "stderr_lines": [], "stdout": "true", "stdout_lines": ["true"]}

2020-06-09 16:47:01.408295 Step: set_fact 
task path: /experiments/generic/kubelet_service_kill/kubelet_service_kill_ansible_logic.yml:131
details => {"ansible_facts": {"flag": "Pass"}, "changed": false}

2020-06-09 16:47:01.452724 Step: [Result]: Getting the final result of kubelet-service-kill experiment 
task path: /experiments/generic/kubelet_service_kill/kubelet_service_kill_ansible_logic.yml:134
details => {
    "msg": "kubelet-service-kill experiment has been Passed"
}
included: /utils/runtime/getting_failure_step.yml for 127.0.0.1

2020-06-09 16:47:01.699133 Step: [Failure-Detection]: Recording the name of failed step 
task path: /utils/runtime/getting_failure_step.yml:44
details => {"ansible_facts": {"failStep": "N/A"}, "changed": false}
included: /utils/runtime/update_chaos_result_resource.yml for 127.0.0.1

2020-06-09 16:47:02.126187 Step: [Result]: Update the chaos result CR to reflect EOT (End of Test) 
task path: /utils/runtime/update_chaos_result_resource.yml:23
details => {"changed": true, "checksum": "80dc5a38f7427593cafca41558fa0c0b6ed1ab16", "dest": "/tmp/chaos-result.yml", "gid": 0, "group": "root", "md5sum": "2724798210e50131f509c2a3772ebd12", "mode": "0644", "owner": "root", "size": 295, "src": "/tmp/ansible-tmp-1591721221.87-169863584406122/source", "state": "file", "uid": 0}

2020-06-09 16:47:02.555716 Step: [The END]: Apply the chaos result CR for kubelet-service-kill experiment 
task path: /utils/runtime/update_chaos_result_resource.yml:33
details => {"changed": true, "cmd": "kubectl apply -f /tmp/chaos-result.yml -n litmus", "delta": "0:00:00.274380", "end": "2020-06-09 16:47:02.531343", "failed_when_result": false, "rc": 0, "start": "2020-06-09 16:47:02.256963", "stderr": "", "stderr_lines": [], "stdout": "chaosresult.litmuschaos.io/kubelet-service-kill configured", "stdout_lines": ["chaosresult.litmuschaos.io/kubelet-service-kill configured"]}
META: ran handlers
META: ran handlers

2020-06-09 16:47:02.558361 ************ RELAX, EXPERIMENT ENDS! ************ 
127.0.0.1                  : ok=39   changed=21   unreachable=0    failed=0  
```

File: experiments/generic/kubelet_service_kill/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+## Experiment Metadata
```
Comment (uditgaurav) [inline experiments/generic/kubelet_service_kill/README.md:1]: will modify

File: chaoslib/litmus/kubelet_service_kill/kubelet-service-kill-ds.j2 [hunk 2]
```diff
@@ -0,0 +1,43 @@
+---
+apiVersion: extensions/v1beta1
```
Comment (ksatchit) [inline chaoslib/litmus/kubelet_service_kill/kubelet-service-kill-ds.j2:2]: can we use pod

File: chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      when: "run_id is not defined or run_id == ''"
+
+    - name: "[Prepare]: Patch the run_id to kubelet service kill daemonset template"
+      template:
+        src:  /chaoslib/litmus/kubelet_service_kill/kubelet-service-kill-ds.j2
+        dest: /tmp/kubelet-service-kill-ds.yml
+
+    - name: "[Prepare]: Create kubelet service kill Daemonset"
+      shell: >
+        kubectl create -f /tmp/kubelet-service-kill-ds.yml -n {{ a_ns }}
```
Comment (ispeakc0de) [inline chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml:49]: c_ns?

File: chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+      retries: 90
+
+  rescue: 
+
+    - name: "[CleanUP]: Tear down service kill infra"
+      shell: >
+        kubectl delete ds -n {{ a_ns }} -l name=service-kill-ds-{{ run_id }}
+      args:
+        executable: /bin/bash
+      register: result
```
Comment (ksatchit) [inline chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml:104]: condition to check if already removed? refer to other experiments for pattern

File: chaoslib/litmus/kubelet_service_kill/service_chaos.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      vars:
+        stage: "ChaosInject"
+        exp_pod_name: "{{ chaos_pod_name }}"
+        engine_ns: "{{ c_ns }}"
+        message: "Injecting {{ c_experiment }} chaos on {{ app_node }} node"
+      when: "c_engine is defined and c_engine != ''"    
+
+    - name: "[Inject]: Killing kubelet service for {{ c_duration }} seconds"
+      shell: >
+        timeout 10 kubectl exec {{ ds_pod }} -n {{ a_ns }}
```
Comment (ksatchit) [inline chaoslib/litmus/kubelet_service_kill/service_chaos.yml:15]: may change when ds -> pod.

File: chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+          retries: 90
+
+      when: 'helper_pod_creation.rc == 0'
+
+
+  rescue: 
+
+    - block:
+      
+        - name: "[CleanUP]: Tear down service kill infra"
```
Comment (ksatchit) [inline chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml:126]: should we retain to know failure

Comment (uditgaurav) [reply chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml:139]: kept rescue block with tear down section
refferd the pattern from other experiments.

File: chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+      args:
+        executable: /bin/bash
+      register: node_state
+      until: node_state.stdout == 'Ready'
+      delay: 2
+      retries: 90
+ 
+    - name: "[CleanUP]: Tear down service kill infra"
+      shell: >
+        kubectl delete pod -n {{ a_ns }} -l name=service-kill-{{ run_id }}
```
Comment (ispeakc0de) [inline chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml:103]: Can we use the file to delete the chaos pod?

File: chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+      shell: >
+        kubectl get pod -l name=service-kill-{{ run_id }} --no-headers -o custom-columns=:status.phase -n {{ a_ns }} | sort | uniq
+      args: 
+        executable: /bin/bash
+      register: result_status
+      until: result_status.stdout ==''
+      delay: 2
+      retries: 90
+ 
+    - name: "[CleanUP]: Tear down service kill infra"
```
Comment (ispeakc0de) [inline chaoslib/litmus/kubelet_service_kill/kubelet_service_kill.yml:118]: redundant task?

**Pull request merged**